### PR TITLE
feat: update knowledge distillation tutorial for using vllm with Qwen model

### PR DESCRIPTION
**Pull request by RexBearIU**
# Description

This pull request significantly updates and modernizes the knowledge distillation tutorial for MaxText, aligning it with current best practices and tooling. The guide now uses Qwen3-32B as the teacher model (via vLLM) and Llama-3.1-8B as the student, streamlines the setup with Hyperdisk storage, and provides new scripts and commands for dataset generation and fine-tuning. The instructions have been clarified, unnecessary conversion steps removed for the teacher, and the fine-tuning process updated for the latest MaxText and vLLM workflows.

# Tests

Manually triggered the distillation pipeline and monitored the execution flow step-by-step. Confirmed that the training loop finished and resources were released. 
image

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [X] I have performed a self-review of my code. For an optional AI review, add the `gemini-review` label.
- [X] I have necessary comments in my code, particularly in hard-to-understand areas.
- [X] I have run end-to-end tests tests and provided workload links above if applicable.
- [X] I have made or will make corresponding changes to the doc if needed, including adding new documentation pages to the relevant Table of Contents (toctree directive) as explained in [our documentation](https://maxtext.readthedocs.io/en/latest/development.html#adding-new-documentation-files).

### CODE REVIEW

File: docs/tutorials/posttraining/knowledge_distillation.md [hunk 1]
```diff
@@ -22,145 +22,238 @@ Knowledge Distillation is a compression technique that transfers knowledge from
...
 #### a. Setup environment variables
 
 ```bash
-export HF_TOKEN = <Hugging Face access token>
-export BASE_DIRECTORY = <Directory to store distillation results>
-export HF_REPO_NAME = <Hugging Face repository name to store teacher-generated dataset>
-export USERNAME_OR_ORG = <Owner of Hugging Face repository>
-export RUN_NAME = <unique name for the run>
+export HF_TOKEN=<your-hf-token> # e.g., hf_BA6...
+export BASE_DIRECTORY=<your-base-directory>  # e.g., knowledge-distillation
```
Comment (SurbhiJainUSC) [inline docs/tutorials/posttraining/knowledge_distillation.md]: nit: `# e.g., gs://`

File: docs/tutorials/posttraining/knowledge_distillation.md [hunk 2]
```diff
@@ -22,145 +22,238 @@ Knowledge Distillation is a compression technique that transfers knowledge from
...
-```sh
-git clone https://github.com/AI-Hypercomputer/maxtext.git
-python3 -m venv ~/venv-maxtext
-source ~/venv-maxtext/bin/activate
-python3 -m pip install uv
-cd maxtext
-uv pip install -r dependencies/requirements/requirements.txt
+To install MaxText and its dependencies for post-training (including vLLM for the teacher), run the following:
+
+1. Follow the [MaxText installation instructions](https://maxtext.readthedocs.io/en/latest/install_maxtext.html#install-maxtext):
```
Comment (SurbhiJainUSC) [inline docs/tutorials/posttraining/knowledge_distillation.md]: nit: `maxtext).`

File: docs/tutorials/posttraining/knowledge_distillation.md [hunk 3]
```diff
@@ -22,145 +22,238 @@ Knowledge Distillation is a compression technique that transfers knowledge from
...
+    hf_access_token=${HF_TOKEN} \
+    base_output_directory=${PRE_TRAINED_MODEL_CKPT_DIRECTORY} \
+    scan_layers=True skip_jax_distributed_system=True
+```
+
+### 3. Generate dataset using vLLM (Teacher Step)
+
+We will use vLLM to generate the dataset from the teacher model.
+
+Create a python script named `generate_distillation_data_vllm.py` with the following content (this script writes a Parquet dataset compatible with MaxText SFT):
```
Comment (SurbhiJainUSC) [inline docs/tutorials/posttraining/knowledge_distillation.md]: Can we make this script available in MaxText like this: https://github.com/AI-Hypercomputer/maxtext/blob/main/tools/data_generation/generate_distillation_data.py

File: docs/tutorials/posttraining/knowledge_distillation.md (hunk 1 above)
Comment (RexBearIU) [reply docs/tutorials/posttraining/knowledge_distillation.md]: Thanks. We use a mounted Hyperdisk because writing large model files and many small I/O ops directly to gs:// is often much slower. The tutorial writes to /mnt/hyperdisk for performance and reproducibility, and I fixed the duplicated env export in the doc.

File: docs/tutorials/posttraining/knowledge_distillation.md (hunk 3 above)
Comment (RexBearIU) [reply docs/tutorials/posttraining/knowledge_distillation.md]: Agreed. I’ve already created the script.

File: tools/data_generation/generate_distillation_data_vllm.py [hunk 4]
```diff
@@ -0,0 +1,204 @@
+# Copyright 2023–2025 Google LLC
```
Comment (SurbhiJainUSC) [inline tools/data_generation/generate_distillation_data_vllm.py]: nit: `2023-2026`

File: tools/data_generation/generate_distillation_data_vllm.py [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+  )
+
+  print("Running inference...")
+  outputs = llm.generate(prompts, sampling_params)
+
+  # Collect results and save directly to Parquet.
+  results = []
+  for output, original_item in zip(outputs, dataset):
+    msgs = list(original_item[PROMPT_COLUMN])
+    msgs.append({"role": "assistant", "content": output.outputs[0].text})
```
Comment (SurbhiJainUSC) [inline tools/data_generation/generate_distillation_data_vllm.py]: You can currently ignoring `num-generations`. Can we fix that?

File: docs/tutorials/posttraining/knowledge_distillation.md [hunk 6]
```diff
@@ -19,148 +19,168 @@
...
 #### a. Setup environment variables
 
 ```bash
-export HF_TOKEN = <Hugging Face access token>
-export BASE_DIRECTORY = <Directory to store distillation results>
-export HF_REPO_NAME = <Hugging Face repository name to store teacher-generated dataset>
-export USERNAME_OR_ORG = <Owner of Hugging Face repository>
-export RUN_NAME = <unique name for the run>
+export HF_TOKEN=<your-hf-token> # e.g., hf_BA6...
+export BASE_NAME=<your-base-directory>  # e.g., knowledge-distillation
```
Comment (SurbhiJainUSC) [inline docs/tutorials/posttraining/knowledge_distillation.md]: Move `BASE_NAME` to the next section where you create the output directory:
```
export BASE_DIRECTORY=/mnt/hyperdisk/${BASE_NAME}
mkdir -p ${BASE_DIRECTORY}
```

File: docs/tutorials/posttraining/knowledge_distillation.md [hunk 7]
```diff
@@ -19,148 +19,168 @@
...
+export BASE_DIRECTORY=/mnt/hyperdisk/${BASE_NAME}
+mkdir -p ${BASE_DIRECTORY}
 ```
 
-### 3. Generate dataset using the teacher model
-Once the teacher model's checkpoint is in the MaxText format, you can run inference to generate the dataset that will be used to fine-tune the student model.
+> **Note:** This tutorial uses a mounted Hyperdisk for performance and reproducibility, because writing large model files and many small I/O operations directly to `gs://` can be significantly slower.
 
-### 3.a. Run the JetStream server
+### 1. Obtain and prepare the teacher model
```
Comment (SurbhiJainUSC) [inline docs/tutorials/posttraining/knowledge_distillation.md]: nit: Remove numbers `1`, `2` from this and following sections.

File: tools/data_generation/generate_distillation_data_vllm.py (hunk 5 above)
Comment (RexBearIU) [reply tools/data_generation/generate_distillation_data_vllm.py]: Thanks for pointing that out! I've fixed it to handle multiple generations.
